cluster/addons.go

```diff
@@ -377,3 +433,69 @@ func (c *Cluster) deployIngress(ctx context.Context) error {
 	log.Infof(ctx, "[ingress] ingress controller %s deployed successfully", c.Ingress.Provider)
 	return nil
 }
+func (c *Cluster) removeDNSProvider(ctx context.Context, dnsprovider string) error {
+	AddonJobExists, err := addons.AddonJobExists(getAddonResourceName(dnsprovider)+"-deploy-job", c.LocalKubeConfigPath, c.K8sWrapTransport)
+	if err != nil {
+		return err
+	}
+	if AddonJobExists {
+		log.Infof(ctx, "[dns] removing DNS provider %s", dnsprovider)
+		if err := c.doAddonDelete(ctx, getAddonResourceName(dnsprovider), false); err != nil {
+			return err
+		}
+
+		log.Infof(ctx, "[dns] DNS provider %s removed successfully", dnsprovider)
+		return nil
+	}
+	return nil
+}
+
+func (c *Cluster) deployDNS(ctx context.Context) error {
+	switch DNSProvider := c.DNS.Provider; DNSProvider {
+	case DefaultDNSProvider:
+		for _, dnsprovider := range DNSProviders {
+			if strings.EqualFold(dnsprovider, DefaultDNSProvider) {
+				continue
+			}
+			if err := c.removeDNSProvider(ctx, dnsprovider); err != nil {
```

Comment:
Do you think it would be easier to do this once before the switch ?
```
       for _, dnsprovider := range DNSProviders {
		if strings.EqualFold(dnsprovider, c.DNS.Provider) {
			continue
		}
		if err := c.removeDNSProvider(ctx, dnsprovider); err != nil {
			return err
		}
	}
```